api/app/db/models/auto_spatial_advisory.py

```diff
@@ -175,20 +171,35 @@ class AdvisoryElevationStats(Base):
 
 
 class AdvisoryFuelStats(Base):
-    """ 
+    """
     Summary statistics about the fuel type of area with high hfi (4k-10k and >10k) per fire shape
     based on the set run_type, for_date and run_datetime.
     """
-    __tablename__ = 'advisory_fuel_stats'
-    __table_args__ = (
-        UniqueConstraint('advisory_shape_id', 'threshold', 'run_parameters', 'fuel_type'),
-        {
-            'comment': 'Fuel type stats per fire shape by advisory threshold'
-        }
-    )
+
+    __tablename__ = "advisory_fuel_stats"
+    __table_args__ = (UniqueConstraint("advisory_shape_id", "threshold", "run_parameters", "fuel_type"), {"comment": "Fuel type stats per fire shape by advisory threshold"})
     id = Column(Integer, primary_key=True, index=True)
     advisory_shape_id = Column(Integer, ForeignKey(Shape.id), nullable=False, index=True)
     threshold = Column(Integer, ForeignKey(HfiClassificationThreshold.id), nullable=False)
     run_parameters = Column(Integer, ForeignKey(RunParameters.id), nullable=False, index=True)
     fuel_type = Column(Integer, ForeignKey(SFMSFuelType.id), nullable=False, index=True)
     area = Column(Float, nullable=False)
+
+
+class AdvisoryTPIStats(Base):
+    """
+    Summary statistics about the elevation based on a HFI >4k classified Topographic Position Index (TPI).
+    Classification of the TPI are bucketed into 1 = valley bottom, 2 = mid slope, 3 = upper slope.
+    Each record includes the raster pixel count of the above classifications as well as the raster pixel size and resolution in metres
+    and an advisory shape the stats are related to.
+    """
+
+    __tablename__ = "advisory_tpi_stats"
+    __table_args__ = {"comment": "Elevation TPI stats per fire shape"}
+    id = Column(Integer, primary_key=True, index=True)
+    advisory_shape_id = Column(Integer, ForeignKey(Shape.id), nullable=False, index=True)
+    run_parameters = Column(Integer, ForeignKey(RunParameters.id), nullable=False, index=True)
+    valley_bottom = Column(Integer, nullable=False, index=True)
```

Comment:
I don't think we need indexes on the valley bottom, mid slope, upper slope and pixel_size_metres columns.

Reply:
Done in: https://github.com/bcgov/wps/pull/3800/commits/31ed215e807ef4e7fbb5e4d483ca53b06d785cad